third_party/xkcp/update_main_license.patch

```diff
@@ -0,0 +1,31 @@
+diff --git a/LICENSE b/LICENSE
+index a7146a8..591ed4d 100644
+--- a/LICENSE
++++ b/LICENSE
+@@ -100,26 +100,6 @@ For XKCP/Standalone/CompactFIPS202/C/genKAT.c, XKCP/tests/UnitTests/genKAT.c and
+     SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ 
+ 
+-For XKCP/support/Kernel-PMU/enable_arm_pmu.c (*not* used in libXKCP, UnitTests, Benchmarks nor KeccakSum):
```

Comment:
I don't think we should be stripping the license. Was there a reason for this?

Reply:
On second thought, I'm guessing we're only using stuff contained in `libXKCP, UnitTests, Benchmarks nor KeccakSum`?

I'd probably call out the reasoning in a comment in the BUILD file that applies this patch.

Reply:
Commenting sounds reasonable. 

I think the bigger issue here is that I do not want and do not need to bring GPL-licensed parts of XKCP. I just could not figure out how to bring some parts XKCP library (without patching). This was easier with vendoring tool by exclusions.

The patching actually ends up bringing all the files to OT repo (see the content of `remove_restricted_files.patch`). Should I be using another tool to create and apply patches?

Reply:
[.](https://github.com/lowRISC/opentitan/pull/16906#issuecomment-1384027314)